pkg/restapi/issuer/operation/operations.go

```diff
@@ -464,14 +471,50 @@ func (c *Operation) preAuthorize(w http.ResponseWriter, r *http.Request) {
 		initiateReq.UserPinRequired = false
 	}
 
-	c.buildInitiateOIDC4CIFlowPage(w, profile.ID, initiateReq, c.preAuthorizeHTML)
+	c.buildInitiateOIDC4CIFlowPage(w, profile.ID, initiateReq, c.preAuthorizeHTML, "")
+}
+
+func (c *Operation) openID4CIHandler(w http.ResponseWriter, r *http.Request) {
+	profile, err := c.determineProfile(r.URL.Query())
+	if err != nil {
+		c.writeErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	initiateReq := &initiateOIDC4CIRequest{}
+
+	flowType := c.determineFlowType(r.URL.Query())
+
+	if flowType == "pre-authorized" {
+		initiateReq = &initiateOIDC4CIRequest{
+			CredentialTemplateID: profile.CredentialTemplateID,
+			UserPinRequired:      true,
+			ClaimData:            &profile.Claims,
+		}
+
+		if strings.EqualFold(r.URL.Query().Get("require_pin"), "false") {
+			initiateReq.UserPinRequired = false
+		}
+	} else if flowType == "authorization" {
```

Comment:
Nevermind - I see that `determineFlowType` always sets one of the two. Note that you could use `defaultFlowType` instead of the string literal `"pre-authorized"` above